internal/host/hostcommands/tang_connectivity_check_cmd.go

```diff
@@ -42,16 +56,60 @@ func (c *tangConnectivityCheckCmd) GetSteps(ctx context.Context, host *models.Ho
 		swag.StringValue(cluster.DiskEncryption.EnableOn) == models.DiskEncryptionEnableOnNone ||
 		swag.StringValue(cluster.DiskEncryption.Mode) == models.DiskEncryptionModeTpmv2 ||
 		!hostutil.IsDiskEncryptionEnabledForRole(*cluster.DiskEncryption, common.GetEffectiveRole(host)) {
-		c.log.Debugf("skipping tangConnectivityCheck for host %s, DiskEncryption config do not require validation here", host.ID.String())
-		return []*models.Step{}, nil
+		c.log.Debugf("skipping tangConnectivityCheck for host %s, cluster DiskEncryption config do not require validation here",
+			host.ID.String())
+		return false
 	}
+	return true
+}
 
+func (c *tangConnectivityCheckCmd) getTangServersFromCluster(cluster common.Cluster, host *models.Host) ([]byte, error) {
+	var (
+		tangServers []byte
+		err         error
+	)
 	request := models.TangConnectivityRequest{TangServers: &cluster.DiskEncryption.TangServers}
-	tangServers, err := json.Marshal(request)
+	tangServers, err = json.Marshal(request)
 	if err != nil {
 		c.log.WithError(err).Errorf("failed to Marshal TangConnectivityRequest with for cluster %s", host.ClusterID)
 		return nil, err
 	}
+	return tangServers, nil
+}
+
+func (c *tangConnectivityCheckCmd) GetSteps(ctx context.Context, host *models.Host) ([]*models.Step, error) {
+	var (
+		cluster     common.Cluster
+		tangServers []byte
+		err         error
+	)
+
+	if err = c.db.First(&cluster, "id = ?", host.ClusterID).Error; err != nil {
+		c.log.WithError(err).Errorf("failed to fetch cluster %s", host.ClusterID)
+		return nil, err
+	}
+	if swag.BoolValue(cluster.Imported) { // Day 2 imported cluster
+		// In this case, it is assumed that the cluster was imported, and thus the DB will have no information
+		// about DiskEncryption settings. Therefore, try to read those settings from the host ignition.
+		if host.APIVipConnectivity != "" {
```

Comment:
Do we need this check? `getTangServersFromHostIgnition` also checks this

Reply:
removed.